src/nvim/window.c

```diff
@@ -5221,11 +5221,17 @@ void win_setheight_win(int height, win_T *win)
     // line, clear it.
     if (full_screen && msg_scrolled == 0 && row < cmdline_row) {
       grid_fill(&default_grid, row, cmdline_row, 0, Columns, ' ', ' ', 0);
+      if (msg_grid.chars) {
+        clear_cmdline = true;
+      }
     }
     cmdline_row = row;
+    p_ch = MAX(Rows - cmdline_row, 1);
+    curtab->tp_ch_used = p_ch;
     msg_row = row;
     msg_col = 0;
     redraw_all_later(NOT_VALID);
+    showmode();
```

Comment:
would setting `redraw_cmdline = true` also work? Then cmdline will be drawn as part of the entire screen

Reply:
Test fails even if I do that

Reply:
hmm you mean the code is broken, but this change makes no difference?

Reply:
The test could be broken as well honestly, but yeah, the change makes no difference